lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -152,29 +154,61 @@ public void OnNext(IAllocatedEvaluator allocatedEvaluator)
         }
 
         /// <summary>
-        /// Specifies the Map or Update task to run on the active context
+        /// Adds active context to _activeContexts collection. After all the active context is received, calls SubmitTasks().
         /// </summary>
         /// <param name="activeContext"></param>
         public void OnNext(IActiveContext activeContext)
         {
-            Logger.Log(Level.Verbose, string.Format("Received Active Context {0}", activeContext.Id));
+            Logger.Log(Level.Verbose, string.Format(CultureInfo.InvariantCulture, "Received Active Context {0}", activeContext.Id));
 
-            if (_serviceAndContextConfigurationProvider.IsMasterEvaluatorId(activeContext.EvaluatorId))
+            lock (_lock)
             {
-                Logger.Log(Level.Verbose, "Submitting master task");
-                _commGroup.AddTask(IMRUConstants.UpdateTaskName);
-                _groupCommTaskStarter.QueueTask(GetUpdateTaskConfiguration(), activeContext);
-                RequestMapEvaluators(_totalMappers);
+                if (_activeContexts.ContainsKey(activeContext.Id))
+                {
+                    var msg = string.Format(CultureInfo.InvariantCulture, "The context [{0}] received is already exists.", activeContext.Id);
```

Comment:
`The context [{0}] received is already exists.` --> `The context [{0}] received already exists.`